src/schema/mod.rs

```diff
@@ -180,42 +183,81 @@ impl FederationSchema {
 
 /// A GraphQL schema with federation data that is known to be valid, and cheap to clone.
 #[derive(Debug, Clone)]
-pub struct ValidFederationSchema(pub(crate) Arc<Valid<FederationSchema>>);
+pub struct ValidFederationSchema {
+    schema: Arc<Valid<FederationSchema>>,
+    subgraph_metadata: Option<SubgraphMetadata>,
```

Comment:
Oh, this is also an issue, `SubgraphMetadata` is not cheap to clone but is part of `ValidFederationSchema` which (in theory) is.

I think we need to tweak the order of initialisation to fix this and maybe let SubgraphMetadata work on not-necessarily-valid federation schemas.